### DISCUSSION TITLE
Solution for markets without a market price

### DISCUSSION THREAD

**Discussion by HenrikJannsen** [bisq-network/bisq2#2310] [General]
Bisq2 only lists markets if we have a market price provided by the price nodes.
The market price is a strong dependency in the whole trade domain and avoiding that would come with a major effort.
To add new markets to the price node require that we have some reliable price provider, which is often not the case for niche markets.

So far I do not see any good solution for that.
Any ideas?

**Comment by suddenwhipvapor** [comment]
"The price provided for this currency is only based on xyz, which could be unreliable and subject to variations that will favor one trade peer at the expense of the other. Please take into account this unavoidable factor when trading in this currency, as Bisq will not be responsible for the aforementioned price variations."
Does it look good enough as a disclaimer that is shown to users who select one of those specific markets?

Or, it can be dealt with a manually set price, like it happens for BSQ swaps on Bisq1, but that is more involved than just adding a warning message like above.

**Reply by HenrikJannsen** [reply]
BSQ price is derived from the trade statistics. We don't have that in Bisq2. 
A disclaimer does not solve the problem that for niche markets there might be no price feed available at all.
But even if, it comes with the effort of implementing the price feed and deploying them into the price nodes, which might not justify the expected trade volume on niche markets.

**Comment by kciouv** [comment]
@HenrikJannsen Why not peg niche currencies to USD or Euro (or some converted averaged value of multiple fiat pairs)? It allows for a fair baseline price to exist without having existing data about the {currency}-{cryptocurrency} pair, which would allow Bisq to bootstrap new markets rather than relying on preexisting centralized ones, and there exists either a USD or Euro pair for every currency under the sun, regardless of political situation.

While it might create situations of Bisq price being subtly different than centralized exchanges, the general reduction in middlemen (and the price pressure their withdrawal fees and other bureaucracy creates) would likely explain the majority of it, and what's left traders could explain in the same way traders explain any markup (or mark down).

**Reply by HenrikJannsen** [reply]
Yes that might be one option, but it comes with a few issues:
- It still creates effort to create the currency-to-USD rate. We need to get a reliable exchange rate provider for that.
- Some of those niche markets might be subject to governmental exchange rate control and therefor official rate might be useless (as is the case for Argentina and Venezuela).
- BTC rates in such countries might come with a quite heavy markup due the lack of local exchanges.
- We would need to mark clearly that those rates are not the same nature as the other rates which are based on aggregated CEX rates.

All that is doable and your suggestion solves the problem of markets where there are no CEX rates available.
But the main question is: how much effort can we sacrifice or such niche markets. A one week dev effort would never get paid off by any potential revenue from those markets.  

I would love to find a really easy solution.
One would be to just allow fixed prices not floating prices, but unfortunately that still would require a lot of code change as currently the market price is expected to exist (should be changed anyway - but too many other higher prio tasks on the table).

**Reply by kciouv** [reply]
@HenrikJannsen Assuming a workable fiat exchange rate provider was found, there could be a separate tab in the interface for "Other currencies," where people could list pairs by ISO 4217 abbreviations and query them by text, without requiring any larger development effort (flag symbols, i8n, whatever). There wouldn't be any need for individual development effort per-currency then, and it would in the long term allow a sane path (market validation) to adding proper support to these currencies once they'd reached an amount of listings that showed it was likely to pay off with resulting fees (once other protocols get implemented, or if there was enough reputation burned for those pairs to cover it) over a year's time.
